lib/spack/spack/solver/asp.py

```diff
@@ -582,9 +567,10 @@ def visit(node):
                     for term in node.body:
                         if ast_type(term) == ASTType.Literal:
                             if ast_type(term.atom) == ASTType.SymbolicAtom:
-                                if ast_sym(term.atom).name == "error":
+                                name = ast_sym(term.atom).name
+                                if name == 'internal_error':
                                     arg = ast_sym(ast_sym(term.atom).arguments[0])
-                                    self.fact(fn.error(arg.string), assumption=True)
+                                    self.fact(getattr(fn, name)(arg.string))
```

Comment:
I think this is clearer here:

```suggestion
                                    self.fact(AspFunction(name)(arg.string))
```